### PR TITLE
checkHeartbeat running every second independently of incoming sensor data (CU-p34wcw)

### PR DESCRIPTION
**Pull request by mariocimet**
This will result in us getting notifications when they go down, not just when they come back on.

### CODE REVIEW

**Comment by mariocimet** [comment]
tested this with a location on dev, it's up there if you want to disconnect and reconnect your XeThrus to check.

File: index.js
Comment (trisapeace) [inline index.js:477]: Should this comment be removed as well?

File: test/serverTest.js
Comment (trisapeace) [inline test/serverTest.js:162]: What are the new `sleep` s for?

**Review by trisapeace — COMMENTED** [review]
I tried this with mine, but didn't get any notifications.

File: test/serverTest.js
Comment (mariocimet) [reply test/serverTest.js:162]: my tests hang locally exactly in between these two, and if solving w a sleep i figured it's preferable to be one-offs rather than in the before/after eachs

File: index.js
Comment (mariocimet) [reply index.js]: yuppers

File: test/serverTest.js
Comment (trisapeace) [reply test/serverTest.js:162]: OK, that's too bad, but if it works, then it works :+1:

File: index.js
Comment (trisapeace) [inline index.js:657]: Is this once a second? That seems extreme.

Comment (mariocimet) [reply index.js:657]: yeah. The actual alert threshold is 60 seconds right now, we could do the same for the interval that we check at? or 30 seconds, any suggestion?

this is also the same interval as for Buttons

Comment (trisapeace) [reply index.js:657]: Ah! That makes sense about the threshold being different than how often we check.

I guess we don't want the maximum time it could be down to be longer than our stillness threshold (which is 60 seconds, right?). So that's probably why we did it this way. What if we did this check every 15 seconds and changed the alert threshold to 45 seconds?

Good point about Buttons. Maybe it's overkill there too, but it does mean that we get more accurate numbers for `heartbeatDashboard` and we have fewer things to check on than we're hopefully going to have for Sensors.

It's up to you :smile:

**Review by trisapeace — APPROVED** [review]
I suggest you wait until after the linting has been merged before merging this.

**Comment by trisapeace** [comment]
Can you please update the CHANGELOG and put the ClickUp number in the PR name for this one as well?
Thanks!

**Comment by trisapeace** [comment]
Task linked: [CU-p34wcw alter heartbeat so that it actually notifies when something goes down, not when it comes back on](https://app.clickup.com/t/p34wcw)

**Pull request merged**
